src/components/panel/panel.js

```diff
@@ -544,4 +651,188 @@ MdPanelRef.prototype._createPanel = function() {
+ * desired properties, then pass the object as part of $mdPanel creation.
+ *
+ * Example:
+ *
+ * var panelPosition = new MdPanelPosition()
+ *     .relativeTo(myButtonEl)
+ *     .withPanelXPosition('center')
+ *     .withPanelYPosition('align-tops');
+ *
+ * $mdPanel.create({
+ *   position: panelPosition
+ * });
+ *
+ * @final @constructor
+ */
+function MdPanelPosition() {
+  /** @private {boolean} */
+  this._absolute = false;
+
+  /** @private {string|undefined} */
+  this._top;
+
+  /** @private {string|undefined} */
+  this._bottom;
+
+  /** @private {string|undefined} */
+  this._left;
+
+  /** @private {string|undefined} */
+  this._right;
+}
+
+
+/**
+ * Sets absolute positioning for the panel.
+ * @return {!MdPanelPosition}
+ */
+MdPanelPosition.prototype.absolute = function() {
+  this._absolute = true;
+  return this;
+};
+
+
+/**
+ * Returns whether the panel should be absolutely positioned.
+ * @returns {boolean}
+ */
+MdPanelPosition.prototype.getAbsolute = function() {
+  return this._absolute;
+};
+
+
+/**
+ * Sets the value of `top` for the panel.
+ * @param {string=} opt_top Value of `top`. Defaults to '0'.
+ * @returns {MdPanelPosition}
+ */
+MdPanelPosition.prototype.top = function(opt_top) {
+  this._top = opt_top || '0';
+  return this;
+};
+
+
+/**
+ * Gets the value of `top` for the panel.
+ * @returns {string}
+ */
+MdPanelPosition.prototype.getTop = function() {
+  return this._top;
+};
+
+
+/**
+ * Sets the value of `bottom` for the panel.
+ * @param {string=} opt_bottom Value of `bottom`. Defaults to '0'.
+ * @returns {MdPanelPosition}
+ */
+MdPanelPosition.prototype.bottom = function(opt_bottom) {
+  this._bottom = opt_bottom || '0';
+  return this;
+};
+
+
+/**
+ * Gets the value of `bottom` for the panel.
+ * @returns {string}
+ */
+MdPanelPosition.prototype.getBottom = function() {
+  return this._bottom;
+};
+
+
+/**
+ * Sets the value of `left` for the panel.
+ * @param {string=} opt_left Value of `left`. Defaults to '0'.
+ * @returns {MdPanelPosition}
+ */
+MdPanelPosition.prototype.left = function(opt_left) {
+  this._left = opt_left || '0';
+  return this;
+};
+
+
+/**
+ * Gets the value of `left` for the panel.
+ * @returns {string}
+ */
+MdPanelPosition.prototype.getLeft = function() {
+  return this._left;
+};
+
+
+/**
+ * Sets the value of `right` for the panel.
+ * @param {string=} opt_right Value of `right`. Defaults to '0'.
+ * @returns {MdPanelPosition}
+ */
+MdPanelPosition.prototype.right = function(opt_right) {
+  this._right = opt_right || '0';
+  return this;
+};
+
+
+/**
+ * Gets the value of `right` for the panel.
+ * @returns {string}
+ */
+MdPanelPosition.prototype.getRight = function() {
+  return this._right;
+};
+
+
+/**
+ * Centers the panel in the x axis. Note: This is the default option,
+ * so we don't set any CSS values.
+ * @returns {MdPanelPosition}
+ */
+MdPanelPosition.prototype.centerHorizontally = function() {
+  return this;
```

Comment:
Ah, I was thinking of times where you'd want to _change_ a position. Is that something you plan on supporting?